File: example/test_source_instrumented.cpp
#include <iostream>
using namespace std;
class Main {

  public:
    inline static unsigned int __INSTRUM_ID = 1;

  public:
    inline static int a = - 10;
    inline static int b = - 20;

    static void something_else ()
    {
        if (a >= b) {
            if (a < 0) cout << "Fizz Buzz!" << endl;
        }
        if (a >= b) {
            if (a < 0) cout << "Fizz Buzz!" << endl;
        }
    }

    static void main ();
};

void Main :: main ()
{
    __INSTRUM_ID = 2;
    {
        {
            std :: cout << "[LOG] '" << __INSTRUM_ID << "before <if_statement> block in [Main :: main] method" << std :: endl;
            if (a >= b) {
                if (a < 0) cout << "Hello World!" << endl;
            }
        }
        if (a >= b) {
            if (a < 0) cout << "Hello World!" << endl;
        }
    }
}

int main ()
{
    Main :: main ();
}

File: example/test_source_instrumented.py
import os
# python test_source_instrumented.py
import logging

class Main :
    def __init__(self) :
        self.a = - 10
        self.b = - 20

    def something_else(self, args) :
        if self.a >= self.b :
            if self.a < 0 :
                print "Fizz Buzz!"

        if self.a >= self.b :
            if self.a < 0 :
                print "Fizz Buzz!"

    def main(self, args) :
        logging.basicConfig()
        self.iLogger = logging.getLogger("simpleExample")
        self.iLogger.setLevel(logging.DEBUG)
        self.iLogger.info("before <if_statement> block in [Main] class, in {main} method")
        if self.a >= self.b :
            if self.a < 0 :
                print "Hello World!"

        if self.a >= self.b :
            if self.a < 0 :
                print "Hello World!"

Main().main({})
